src/main/java/nextstep/ladder/domain/Ladder.java

```diff
@@ -1,25 +1,29 @@
 package nextstep.ladder.domain;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 public class Ladder {
-  private final Map<LadderPoint, LadderBarStatus> ladderBarStatuses;
-  private final int ladderLength;
-
-  public Ladder(Map<LadderPoint, LadderBarStatus> ladderBarStatuses, int ladderLength) {
-    this.ladderBarStatuses = ladderBarStatuses;
-    this.ladderLength = ladderLength;
+  private final LadderPoints ladderPoints;
+  public Ladder(LadderPoints ladderPoints) {
+    this.ladderPoints = ladderPoints;
   }
 
-  public Map<LadderPoint, LadderBarStatus> getLadderBarStatuses() {
-    return this.ladderBarStatuses;
+  public LadderPoints ladderPoints() {
+    return this.ladderPoints;
   }
 
-  public LadderBarStatus getLadderBarStatus(LadderPoint ladderPoint) {
-    return this.ladderBarStatuses.get(ladderPoint);
-  }
+  public Players play(PlayerNames playerNames, Map<Integer, String> rewards) {
```

Comment:
```suggestion
  public MatchingResult play() {
```
사다리타기를 통해 실행한 결과와 PlayerNames, Rewards를 매핑하는 부분을 분리해 보는 것은 어떨까?

MatchingResult는 Map<Integer, Integer>에 대한 일급콜렉션 객체이다.
Map의 key는 시작 위치, values는 사다리를 실행한 결과 위치이다.

MatchingResult 값을 활용해 PlayerNames, Rewards을 매칭하는 것은 다른 객체로 위임해 보는 것은 어떨까?